### PR TITLE
Fixed ErrorBarPlot mpl2.0 compat and made it more general

### PR DESCRIPTION
**Pull request by philippjfr**
ErrorBars were one of the only failures in our compatibility with mpl 2.0, the difference is that errorbar caps are disabled by default and that the return signature differs. This adds that compatibility and also made the plot correctly animate ErrorBar plots with varying numbers of error bars and adds support for inverting axes on the plot allowing it to be adjoined correctly.

### CODE REVIEW

File: holoviews/plotting/mpl/chart.py [hunk 1]
```diff
@@ -101,7 +107,14 @@ class ErrorPlot(ChartPlot):
     _plot_methods = dict(single='errorbar')
 
     def init_artists(self, ax, plot_data, plot_kwargs):
-        _, (bottoms, tops), verts = ax.errorbar(*plot_data, **plot_kwargs)
+        handles = ax.errorbar(*plot_data, **plot_kwargs)
+        bottoms, tops = None, None
+        if mpl_version >= str('2.0'):
```
Comment (jlstevens) [inline holoviews/plotting/mpl/chart.py]: So this is the only bit that changes specifically for matplotlib 2.0+?

Comment (philippjfr) [inline holoviews/plotting/mpl/chart.py]: No the update code also had to change because the caps aren't instantiated by default.

**Comment by philippjfr** [comment]
Tests appear to be failing because this display output 46 in Elements has changed. Here is what it looks like:

image

What that has to do with errorbars I do not know.

**Comment by jlstevens** [comment]
Wow..that is a pretty strange side-effect! Definitely no error bars there!

**Comment by philippjfr** [comment]
So I'm super confused now, the test above is meant to look like this:

image

Why would annotations disappear after fixing a completely different plot?

**Comment by philippjfr** [comment]
Figured it out, this is why ``*`` imports are bad. The import of ``Path`` from matplotlib is getting imported into ``mpl/__init__.py`` which ends up overriding our registration ``hv.Path``, so paths never end up getting plotted.

**Comment by jlstevens** [comment]
> The import of Path from matplotlib is getting imported into mpl/__init__.py which ends up overriding our registration of hv.Path, so paths never end up getting plotted.

Ha! Yes, let's not use glob imports.

Anyway, now that is resolved, this PR looks fine and the tests have passed without needing to be updated. Merging.

**Pull request merged**